lib/fog/vsphere/requests/compute/create_vm.rb

```diff
@@ -34,9 +39,47 @@ def create_vm attributes = { }
 
         private
 
+        def create_vm_on_datastore(vm_cfg, vmFolder, resource_pool)
+          vm = vmFolder.CreateVM_Task(:config => vm_cfg, :pool => resource_pool).wait_for_completion
+        end
+
+        def create_vm_on_storage_pod(storage_pod, vm_cfg, vmFolder, resource_pool, datacenter)
+          pod_spec     = RbVmomi::VIM::StorageDrsPodSelectionSpec.new(
+            :storagePod => get_raw_storage_pod(storage_pod, datacenter),
+          )
+          storage_spec = RbVmomi::VIM::StoragePlacementSpec.new(
+            :type => 'create',
+            :folder => vmFolder,
+            :resourcePool => resource_pool,
+            :podSelectionSpec => pod_spec,
+            :configSpec => vm_cfg,
+          )
+          srm = @connection.serviceContent.storageResourceManager
```

Comment:
I think you could use the getter here (`connection`). This would make it easier to change the internals without breaking the services if needed.